### DISCUSSION TITLE
exit action gets triggered when FSM is stopped

### DISCUSSION THREAD

**Discussion by goldenratio** [statelyai/xstate#3462] [Q&A]
Hi,
When I call `fsm.stop();`, then current state's `exit` actions are invoked. How can I avoid it?

**Comment by goldenratio** [comment]
we need to wrap the exit function with assign

**Comment by davidkpiano** [comment]
This is expected behavior. When the state machine stops, all states are exited; therefore, all exit actions get executed.

**Reply by goldenratio** [reply]
If we use assign, it doesn't get executed. is that expected?

**Reply by Andarist** [reply]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.
